### ISSUE TITLE
Log message about logging configuration file interferes with bashlib

### ISSUE THREAD

**Issue by kba** [OCR-D/core#667]
With a logging configuration file at `$HOME/ocrd_logging.conf`, this happens:

```
ocrd workspace find -G OCR-D-IMG -k local_filename -k ID -k pageId                                
13:04:27.265 DEBUG ocrd.logging - Picked up logging config at /home/kba/ocrd_logging.conf
OCR-D-IMG/INPUT_0017.tif        INPUT_0017      PHYS_0017
OCR-D-IMG/INPUT_0020.tif        INPUT_0020      PHYS_0020
```

For bashlib processors like ocrd_im6convert, this will lead to an error like:

```
ocrd-im6convert: line 68: fields[1]: unbound variable
```

My first idea would be to make that "self-logging" behavior optional. But to make it optional, we'd have to rely on the configuration file. So if users want to debug logging itself, such as the configuration file not being found, they would have to configure that - in a configuration file :-?

**Comment by bertsky** [comment]
> My first idea would be to make that "self-logging" behavior optional. But to make it optional, we'd have to rely on the configuration file. So if users want to debug logging itself, such as the configuration file not being found, they would have to configure that - in a configuration file :-?

They could still use the command-line override `-l` for that. I'd say remove the self-logging by default.

But... 

> With a logging configuration file at `$HOME/ocrd_logging.conf`, this happens:
> 
> ```
> ocrd workspace find -G OCR-D-IMG -k local_filename -k ID -k pageId                                
> 13:04:27.265 DEBUG ocrd.logging - Picked up logging config at /home/kba/ocrd_logging.conf
> OCR-D-IMG/INPUT_0017.tif        INPUT_0017      PHYS_0017
> OCR-D-IMG/INPUT_0020.tif        INPUT_0020      PHYS_0020
> ```

...I _also_ think we are still not safe against logger output from other events throughout bashlib processors. So perhaps instead of calling `ocrd`, these kinds of functionalities could be wrapped into functions to be sourced via bashlib. So instead of `ocrd workspace find ...` you would then do `ocrd__workspace_find`. Or even do more advanced stuff like `ocrd__processor_input_files`, `ocrd__page OCR-D-SEG-LINE_file_0001 get //PcGts/Page/TextRegion/@id` and `ocrd__page OCR-D-SEG-LINE_file_0001 get_image region0002 feature_filter=binarized fill=white` (which we currently cannot solve satisfiably).

**Comment by kba** [comment]
> So instead of ocrd workspace find ... you would then do ocrd__workspace_find

And we would solve the issue by filtering out lines that look like log message?

> `ocrd__page OCR-D-SEG-LINE_file_0001 get_image region0002 feature_filter=binarized fill=white `

Agreed, though this is best discussed in the context of https://github.com/OCR-D/core/issues/264

**Comment by bertsky** [comment]
> > So instead of ocrd workspace find ... you would then do ocrd__workspace_find
> 
> And we would solve the issue by filtering out lines that look like log message?

No. These functions should be evaluated in a context without any log handlers. Or at least redirect them to another FD.

Speaking of which, we could do that by default within `ocrd__wrap` already, e.g. make the default handler output to stderr instead of stdout.

**Comment by kba** [comment]
> No. These functions should be evaluated in a context without any log handlers. Or at least redirect them to another FD.

That sounds good but I'm not sure how this would work. We do still want to see the log messages produced by core itself or `ocrd log`. And what would be the advantage over `ocrd -l OFF workspace find ...`?

> make the default handler output to stderr instead of stdout.

That might be worthwile to change in the specs and our default logging configuration and configuration file. When we defined the default logging as stdout, that was mostly for ease of implementation (b/c we didn't anticipate that OCR-D/core would be used by all processors).

> I'd say remove the self-logging by default.

Agreed, this is mostly useful for core devs to debug the logging logic but is a PITA for everyone else.

**Comment by bertsky** [comment]
> > No. These functions should be evaluated in a context without any log handlers. Or at least redirect them to another FD.
> 
> That sounds good but I'm not sure how this would work.

Not sure myself what I meant. We don't have any override mechanisms beside config files and `-l OFF`...

> We do still want to see the log messages produced by core itself or `ocrd log`

Right, we don't want to disable logging, just redirect it. So even if we do make stderr the new default, then `ocrd__log` could redirect these to stdout if the bashlib processor wants to mix logging with other output.

> And what would be the advantage over `ocrd -l OFF workspace find ...`?

None, except it might be shorter. In fact, we could define some
```sh
ocrd__workspace () {
    ocrd -l OFF workspace -d "${ocrd__argv[working_dir]}" "$@"
}
```

> > make the default handler output to stderr instead of stdout.
> 
> That might be worthwile to change in the specs and our default logging configuration and configuration file.

This would have also saved us a lot of trouble with `--dump-json`.(And thinking about the "rogue logger" and module-level import problem again, we can to this day not rule that out re-surfacing.)

**Comment by bertsky** [comment]
But if stderr was in place, we would not need `-l OFF` and could actually rely on the backend to show the most appropriate error message. It would also benefit all existing bashlib processors automatically.
